### PR TITLE
OHRM5X-1346: Get the Current Date as Prop

### PR DESCRIPTION
**Pull request by SandeepaOHRM**

### CODE REVIEW

File: src/plugins/orangehrmCorePlugin/Helper/VueControllerHelper.php [hunk 1]
```diff
@@ -251,12 +256,45 @@ public function getBreadcrumb(): array
...
+    /**
+     * @return ConfigDao
+     */
+    private function getConfigDao(): ConfigDao
+    {
+        if (is_null($this->configDao)) {
+            $this->configDao = new ConfigDao();
+        }
+        return $this->configDao;
+    }
```
Comment (RajithaKumara) [inline src/plugins/orangehrmCorePlugin/Helper/VueControllerHelper.php:299]:
```suggestion
```

File: src/plugins/orangehrmCorePlugin/Helper/VueControllerHelper.php [hunk 2]
```diff
@@ -78,6 +80,8 @@ class VueControllerHelper
      */
     protected ?ScreenPermissionService $screenPermissionService = null;
 
+    private ?ConfigDao $configDao = null;
+
```
Comment (RajithaKumara) [inline src/plugins/orangehrmCorePlugin/Helper/VueControllerHelper.php:84]:
```suggestion
```

```php
class VueControllerHelper
{
    use ModuleScreenHelperTrait;
    use UserServiceTrait;
    use MenuServiceTrait;
    use I18NHelperTrait;
    use ConfigServiceTrait; // <-- use this trait
```

File: src/plugins/orangehrmCorePlugin/Helper/VueControllerHelper.php [hunk 3]
```diff
@@ -251,12 +256,45 @@ public function getBreadcrumb(): array
...
         }
         return $breadcrumb;
     }
+
+    /**
+     * @return array
+     */
+    private function getDateFormat(): array
+    {
+        $dateFormat = $this->getConfigDao()->getValue('admin.localization.default_date_format');
```
Comment (RajithaKumara) [inline src/plugins/orangehrmCorePlugin/Helper/VueControllerHelper.php:273]:
```suggestion
        $dateFormat = $this->getConfigService()->getAdminLocalizationDefaultDateFormat();
```

File: src/plugins/orangehrmCorePlugin/Helper/VueControllerHelper.php [hunk 4]
```diff
@@ -251,12 +256,45 @@ public function getBreadcrumb(): array
...
+        switch ($dateFormat) {
+            case 'd-m-Y':
+                $label = 'dd-mm-yyyy';
+                break;
+            case 'm-d-Y':
+                $label = 'mm-dd-yyyy';
+                break;
+            default:
+                $label = 'yyyy-mm-dd';
```
Comment (RajithaKumara) [inline src/plugins/orangehrmCorePlugin/Helper/VueControllerHelper.php:282]: shall we maintain a map (assoc array), rather than large switch-case block

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/plugins/orangehrmCorePlugin/Helper/VueControllerHelper.php (hunk 1 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmCorePlugin/Helper/VueControllerHelper.php:299]: done

File: src/plugins/orangehrmCorePlugin/Helper/VueControllerHelper.php (hunk 2 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmCorePlugin/Helper/VueControllerHelper.php:84]: done

File: src/plugins/orangehrmCorePlugin/Helper/VueControllerHelper.php (hunk 3 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmCorePlugin/Helper/VueControllerHelper.php:273]: done

File: src/plugins/orangehrmCorePlugin/Helper/VueControllerHelper.php (hunk 4 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmCorePlugin/Helper/VueControllerHelper.php]: Yes, replaced with a map

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**